File: redis/redis01_as_db_template/src/main/java/com/sc/redis/controller/EmployeeController.java
package com.sc.redis.controller;

import com.sc.redis.dao.EmployeeDao;
import com.sc.redis.entity.Employee;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/employee")
public class EmployeeController {
    private final EmployeeDao employeeDao;

    public EmployeeController(EmployeeDao employeeDao) {
        this.employeeDao = employeeDao;
    }

    @PostMapping
    public Employee save(@RequestBody Employee employee) {
        return employeeDao.save(employee);
    }

    @GetMapping
    public List<Employee> findAll() {
        return employeeDao.findAll();
    }

    @GetMapping("{id}")
    public Employee findById(@PathVariable Long id) {
        return employeeDao.findById(id);
    }

    @DeleteMapping("{id}")
    public void delete(@PathVariable Long id) {
        employeeDao.delete(id);
    }
}

File: kotlin/kotlin_springboot01/src/main/kotlin/com/sc/kotlin/controller/EmployeeController.kt
package com.sc.kotlin.controller

import com.sc.kotlin.entity.Employee
import com.sc.kotlin.service.EmployeeService
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/employee")
class EmployeeController(
        private val employeeService: EmployeeService
) {
    @GetMapping
    fun findAll(): List<Employee> {
        return employeeService.findAll()
    }

    @PostMapping
    fun save(@RequestBody employee: Employee): Employee {
        return employeeService.save(employee)
    }
}
